staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -185,7 +185,10 @@ func translateServicePortToTargetPort(ports []string, svc corev1.Service, pod co
 			return nil, err
 		}
 
-		if int32(portnum) != containerPort || localPort == "" {
+		// should fail when localPort is empty (=> use random local port)
+		localportnum, err := strconv.Atoi(localPort)
+
+		if int32(portnum) != containerPort || localPort == "" || (int32(localportnum) != containerPort && err == nil) {
```

Comment:
this is significantly harder to follow... it looks like the regression came from https://github.com/kubernetes/kubernetes/pull/85511/files#diff-250be3561ddb6980e7108f04f2d15b86 and came from changing the `else` block...